src/test/scala/firrtlTests/UnitTests.scala

```diff
@@ -321,4 +320,53 @@ class UnitTests extends FirrtlFlatSpec {
       compileToVerilog(input)
     }
   }
+
+  "Parsing SInts" should "work" in {
+    val passes = Seq()
+    val input =
+      """circuit Unit :
+        |  module Unit :
+        |    node posSInt = SInt(13)
+        |    node negSInt = SInt(-13)
+        |    node posSIntString = SInt("h0d")
+        |    node posSIntString2 = SInt("h+d")
+        |    node posSIntString3 = SInt("hd")
+        |    node negSIntString = SInt("h-d")
+        |    node negSIntStringWidth = SInt<5>("h-d")
+        |    node neg2 = SInt("h-2")
+        |    node pos2 = SInt("h2")
+        |""".stripMargin
+    val expected = Seq(
+      """node posSInt = SInt<5>("hd")""",
+      """node negSInt = SInt<5>("h-d")""",
+      """node posSIntString = SInt<5>("hd")""",
+      """node posSIntString2 = SInt<5>("hd")""",
+      """node posSIntString3 = SInt<5>("hd")""",
+      """node negSIntString = SInt<5>("h-d")""",
+      """node negSIntStringWidth = SInt<5>("h-d")""",
+      """node neg2 = SInt<2>("h-2")""",
+      """node pos2 = SInt<3>("h2")"""
+    )
+    executeTest(input, expected, passes)
```

Comment:
Are one of these tests illustrating why the the width of literal calculation needed to change?